pkg/boot/kexec/kexec_load_linux.go

```diff
@@ -65,3 +95,182 @@ func rawLoad(entry uintptr, segments []Segment, flags uint64) error {
+	var kmem *Memory
+
+	bp, err := ioutil.ReadFile("/sys/kernel/boot_params/data")
+	if err != nil {
+		return fmt.Errorf("Reading boot_param data: %w", err)
+	}
+	var lp = &bzimage.LinuxParams{}
+	if err := lp.UnmarshalBinary(bp); err != nil {
+		return fmt.Errorf("Unmarshaling header: %w", err)
+	}
+	Debug("Start LoadBzImage...")
+	Debug("Try decompressing kernel...")
+	kb, err := uio.ReadAll(kernel)
+	if err != nil {
+		return err
+	}
+	Debug("Try parsing bzImage...")
+	if err := b.UnmarshalBinary(kb); err != nil {
+		return err
+	}
+	Debug("Done parsing bzImage.")
+
+	if len(b.KernelCode) < 1024 {
+		return fmt.Errorf("kernel code size smaller than 1024 bytes: %d", len(b.KernelCode))
+	}
+
+	Debug("Try get ELF from bzImage...")
+	kelf, err := b.ELF()
+	if err != nil {
+		return err
+	}
+	Debug("Done.")
+	kernelEntry := uintptr(kelf.Entry)
+	Debug("kernelEntry: %v", kernelEntry)
+
+	// Prepare segments.
+	kmem = &Memory{}
+	Debug("Try parsing memory map...")
+	// TODO(10000TB): refactor this call into initialization of
+	// kexec.Memory, as it does not depend on specific boot.
+	if err := kmem.ParseMemoryMap(); err != nil {
+		return fmt.Errorf("parse memory map: %v", err)
+	}
+
+	var relocatableKernel bool
+	if b.Header.Protocolversion < 0x0205 {
+		return fmt.Errorf("bzImage boot protocol earlier thatn 2.05 is not supported currently: %v", b.Header.Protocolversion)
+	}
+	relocatableKernel = b.Header.RelocatableKernel != 0
+	// Only protected mode is currently supported.
+	// In protected mode, kernel need be relocatable, or it will need to fall
+	// to real mode executing.
+
+	if !relocatableKernel {
+		return errors.New("non-relocateable Kernels are not supported")
+	}
+
+	Debug("Loading purgatory...")
+
+	if _, err := kmem.LoadElfSegments(bytes.NewReader(b.KernelCode)); err != nil {
+		return err
+	}
+
+	var ramfsRange Range
+	if ramfs != nil {
+		b, err := ioutil.ReadAll(ramfs)
+		if err != nil {
+			return fmt.Errorf("unable to read initramfs: %w", err)
+		}
+		if ramfsRange, err = kmem.AddKexecSegment(b); err != nil {
+			return fmt.Errorf("Adding initramfs segment: %v", err)
+		}
+		Debug("Added %d byte initramfs at %s", len(b), ramfsRange)
+		lp.Initrdstart = uint32(ramfsRange.Start)
+		lp.Initrdsize = uint32(ramfsRange.Size)
+	}
+	// TODO(10000TB): Insert cmdline.
+	// cmdlineLen := len(cmdline) + 1
+	// cmdlineR, err := kmem.FindSpace(cmdlineLen)
+	//
+	// Do we just modify the bzImage with the cmdline passed in
+	// Or we need to allocate a new segments for the cmdline
+	// and insert it into segments list to be loaded ?
+	//
+	// kernelCmdlineLen := 0
+
+	var setupRange Range
+	// The kernel is a bzImage kernel if the protocol >= 2.00 and the 0x01
+	// bit (LOAD_HIGH) in the loadflags field is set
+	// TODO(10000TB): check on loadflags.
+	linuxParam, err := lp.MarshalBinary()
+	if err != nil {
+		return fmt.Errorf("Re-marshaling header: %w", err)
+	}
+
+	// TODO(10000TB): Free mem hole start end aligns by
+	// max(16, pagesize).
```

Comment:
https://github.com/u-root/u-root/blob/d29928f4a7dd41f32742fbd87fdb190fd5e2e885/pkg/boot/kexec/memory_linux.go#L633

Write one of these functions but also make the start align (we unfortunately assume that the Ranges are already aligned, which is a *wrong* assumption)

Reply:
And then you can simply add the purgatory (trampoline) like this: https://github.com/u-root/u-root/blob/d29928f4a7dd41f32742fbd87fdb190fd5e2e885/pkg/boot/multiboot/multiboot.go#L511

Reply:
SG. I'll note such in the same TODO. 

I wanted to have a minimal working thing reviewed and submitted, to transition from 0 to 1. 
I am sure I'll fix it in a way that you will like.